src/v/cluster/distributed_kv_stm.h

```diff
@@ -65,7 +67,24 @@ concept SerdeSerializable = requires(T t, iobuf buf, iobuf_parser parser) {
     serde::to_iobuf(t);
 };
 
-template<SerdeSerializable Key, SerdeSerializable Value>
+// Currently we require constexpr static memory usage, but there is no reason we
+// couldn't track this dynamically if we didn't have fixed size keys or values.
+template<class T>
+concept StaticMemoryUsageReporter = requires {
+    // Ensure there is a static function for memory usage
+    { T::memory_usage() } -> std::same_as<size_t>;
+    // Ensure that this is a constant expression, by checking evaluation as a
+    // template parameter.
+    {
+        std::bool_constant<T::memory_usage() >= 0>()
+    } -> std::same_as<std::true_type>;
+};
```

Comment:
wondering if we can enforce this using "is_trivially_copyable" type trait for key and value types, which effectively guarantees static memory usage (and get rid of memory_usage()), unlikely we will ever use dynamic objects as KVs?

Reply:
Sounds good to me! Thanks